apps/server/src/shared/repo/task/task-scope.ts

```diff
@@ -14,9 +14,17 @@ export class TaskScope extends Scope<Task> {
 		return this;
 	}
 
-	byOnlyCreatorId(teacherId: EntityId): TaskScope {
+	byAssignedUser(assignedUserId: EntityId): TaskScope {
 		this.addQuery({
-			$and: [{ creator: teacherId }, { course: null }, { lesson: null }],
+			$or: [{ users: { $exists: false } }, { users: { $in: [assignedUserId] } }],
```

Comment:
what would be the usecase/business case for the `users` array not being present but still geting into the assignment scenario?

Reply:
In case we `users = undefined` we do not want to apply the filter because this ensures backwards compatibility.
Without the `{ users: { $exists: false } }` students could not access the tasks in their groups if they are not listed in `users`.

Students should be filtered out if the `users` array is present if not we want to not filter the student by `users` but by parent (e.g. course).